### PR TITLE
Render markdown code blocks

### PR DESCRIPTION
**Pull request by fmterrorf**
Fixes https://github.com/elixirschool/school_house/issues/187

Before
Screenshot from 2022-03-05 12-46-27

After
Screenshot from 2022-03-05 12-45-53

Before
Screenshot from 2022-03-05 12-47-16

After
Screenshot from 2022-03-05 12-47-58

### CODE REVIEW

**Comment by fmterrorf** [comment]
Oops. I didn't notice that the test is failing. I'm going to fix that shortly

File: lib/school_house_web/templates/post/_post_preview.html.heex [hunk 1]
```diff
@@ -9,8 +9,8 @@
     <% end %>
   </div>
   <%= link(to: Routes.post_path(@conn, :show, @post.slug), class: "block mt-4") do %>
-    <p class="text-xl font-semibold text-primary dark:text-primary-dark"><%= @post.title %></p>
-    <p class="mt-3 text-base text-primary dark:text-primary-dark"><%= @post.excerpt %></p>
+    <span class="text-xl font-semibold text-primary dark:text-primary-dark prose"><%= raw @post.title %></span>
+    <span class="mt-3 text-base text-primary dark:text-primary-dark prose"><%= raw @post.excerpt %></span>
```
Comment (kinson) [inline lib/school_house_web/templates/post/_post_preview.html.heex:13]:
```suggestion
    <span class="text-xl font-semibold prose dark:prose-dark"><%= raw @post.title %></span>
    <span class="mt-3 text-base prose dark:prose-dark"><%= raw @post.excerpt %></span>
```

File: lib/school_house_web/templates/post/post.html.heex [hunk 2]
```diff
@@ -15,14 +15,16 @@
     </div>
 
     <div class="pt-6 text-center">
-      <h1 class="text-2xl leading-8 font-extrabold tracking-tight text-primary dark:text-primary-dark sm:text-2xl"><%= @post.title %></h1>
+      <h1 class="text-2xl leading-8 font-extrabold tracking-tight text-primary dark:text-primary-dark sm:text-2xl prose"><%= raw @post.title %></h1>
```
Comment (kinson) [inline lib/school_house_web/templates/post/post.html.heex:18]:
```suggestion
      <h1 class="text-2xl leading-8 font-extrabold tracking-tight sm:text-2xl prose dark:prose-dark"><%= raw @post.title %></h1>
```

File: lib/school_house_web/templates/post/post.html.heex [hunk 3]
```diff
@@ -15,14 +15,16 @@
...
     </div>
 
     <div class="py-2 text-center">
       <h2 class="text-medium leading-8 tracking-tight text-primary dark:text-primary-dark sm:text-large ">By <%= @post.author %></h2>
     </div>
 
-    <div class="bg-brand-gray-300 dark:bg-brand-gray-800 rounded-xl p-4 mb-3 text-xl text-light dark:text-light-dark sm:mb-4 text-center"><%= raw @post.excerpt %></div>
+    <div class="bg-brand-gray-300 dark:bg-brand-gray-800 rounded-xl p-4 mb-3 text-xl text-light dark:text-light-dark sm:mb-4 text-center prose">
```
Comment (kinson) [inline lib/school_house_web/templates/post/post.html.heex:25]:
```suggestion
    <div class="bg-brand-gray-300 dark:bg-brand-gray-800 rounded-xl p-4 mb-3 text-xl sm:mb-4 text-center prose dark:prose-dark">
```

**Review by kinson — COMMENTED** [review]
Thanks for tackling this, it looks much nicer!

I pulled this down and it looks great in light mode 😎 

I added a few suggestions to get the dark mode version styled in a way that looks good as well. The only thing not left in the suggestions is a few changes to the "dark" section of the tailwindcss config file:

On line 145, I updated the background color for `pre` with:
```
'background-color': theme('colors.brand-gray-800')
```

And on line 166~ I added another block to the `typography.dark` section with:

```
code: {
  color: theme('colors.brand-gray-200'),
  'background-color': theme('colors.brand-gray-800')
}
```

Let me know if you have any questions or have a better solution for accommodating dark mode!

Here are some screenshots of light and dark mode after these changes:
Screen Shot 2022-03-08 at 9 30 26 PM
Screen Shot 2022-03-08 at 9 30 18 PM

**Comment by fmterrorf** [comment]
Thanks for the review @kinson ! Sorry I haven't looked at this PR for a while. I'll try to push in the changes by today

**Comment by doomspork** [comment]
@fmterrorf would you like some help incorporating these changes and getting your work merged? I'd be happy to help as I'm sure is @kinson. 😁

**Review by doomspork — APPROVED** [review]
@fmterrorf I went ahead and accepted @kinson's suggestions so that you don't need to get anything 🎉  Thanks so much for making this fix 🙏

**Comment by fmterrorf** [comment]
> @fmterrorf I went ahead and accepted @kinson's suggestions so that you don't need to get anything tada Thanks so much for making this fix pray

Thanks for pushing it! Sorry i got a little bit busy and forgot about this PR

**Pull request merged**